common/views/pages/_app.tsx

```diff
@@ -45,19 +46,23 @@ function isErrorPage(route: string): boolean {
 
 const dev = process.env.NODE_ENV !== 'production';
 
+export type NextPageWithLayout<P = unknown, IP = P> = NextPage<P, IP> & {
+  getLayout?: (page: ReactElement) => ReactElement;
+};
+
 type GlobalProps = {
   serverData: ServerData;
   pageview?: Pageview;
   gaDimensions?: GaDimensions;
 } & Partial<AppErrorProps>;
 
-type WecoAppProps = Omit<AppProps, 'pageProps'> & { pageProps: GlobalProps };
+type WecoAppProps = Omit<AppProps, 'pageProps'> & {
+  pageProps: GlobalProps;
+  Component: NextPageWithLayout;
+};
 
-const WecoApp: FunctionComponent<WecoAppProps> = ({
-  Component,
-  pageProps,
-  router,
-}) => {
+const WecoApp: FunctionComponent<WecoAppProps> = props => {
+  const { pageProps, router } = props;
```

Comment:
For my knowledge; why change the above, we're still wanting `Component` too below, right?

Reply:
Oh I see we're declaring a different Component afterwards 🤔